### ISSUE TITLE
Single-statement async functions using `withCheckedThrowingContinuation` trigger `implicit_return` rule

### ISSUE THREAD

**Issue by KacperCzapp** [realm/SwiftLint#3781]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

SwiftLint is triggered in single-statement `async` functions that use the `withCheckedThrowingContinuation` helper.

### Environment

* SwiftLint version: 
  * 0.45.1
* Installation method used: 
  * CocoaPods

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  * no
* Which Xcode version are you using (check `xcodebuild -version`)?
  * Xcode 13.2, Build version 13C90
* Do you have a sample that shows the issue?

```swift
func sendCommand() async throws {
    return try await withCheckedThrowingContinuation { continuation in
//  ^ Implicit Return Violation: Prefer implicit returns in closures, functions and getters. 
        sendRequest() { response in
            switch response {
            case .success:
                continuation.resume()
            case .failure(let error):
                continuation.resume(throwing: error)
            }
        }
    }
}
```

If no explicit `return` is used, the compiler then says `Generic parameter 'T' could not be inferred`.

**Comment by KacperCzapp** [comment]
Weird, the error went away after some time by itself. Closing now.

**Comment by KacperCzapp** [comment]
Reopening, I was fooled by the "Superflous Disable Command". 

Which is weird... Disabling the `implicit_return` using `//swiftlint:disable:this implicit_return` causes "Superflous Disable" warning, yet after removing it Swiftlint is still triggered by it 🤔

**Comment by KacperCzapp** [comment]
Could anyone take a look at this? I'm encountering it now in many places.

**Comment by jpsim** [comment]
The implicit return rule is functioning as expected here, it's reporting a violation where there's an explicit `return` statement that could be removed. In this case when you remove the explicit return, however, you need to help out the compiler's type inference engine a bit to give it more context. For example:

```swift
func sendRequest(response: (Result<Void, Error>) -> Void) {
  response(.success(()))
}

func sendCommand() async throws {
  try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
    sendRequest { response in
      switch response {
      case .success:
        continuation.resume()
      case .failure(let error):
        continuation.resume(throwing: error)
      }
    }
  }
}
```

Since I don't have the source of your `sendRequest()` function I came up with one here that satisfies the rest of your example, but your types may vay.
